enhancements/ingress/admin-configured-ca-bundle.md

```diff
@@ -0,0 +1,240 @@
+api-approvers:
+  - None
+creation-date: 2023-11-02
+last-updated: 2023-12-01
+tracking-link:
+  - https://issues.redhat.com/browse/RFE-2182
+  - https://issues.redhat.com/browse/OCPSTRAT-431
+  - https://issues.redhat.com/browse/NE-761
+see-also:
+  - None
+replaces:
+  - None
+superseded-by:
+  - None
+---
+
+# Support for admin configured CA trust bundle in Ingress Operator
+
+## Summary
+
+This enhancement describes the proposal to allow an administrator of OpenShift to configure
+trusted Certificate Authorities (CA) to verify the services' certificate by OpenShift Router
+for the re-encrypted traffic.
+
+## Motivation
+
+When an OpenShift administrator creates a route with the re-encrypt termination type, they
+have to also provide a CA certificate for that route if it uses a custom CA instead of the
+default OpenShift CA for certificate validation. In addition, when the same CA is used by the
+administrator for signing several services' certificates it is time-consuming and error-prone
+to configure the same CA certificate in every created route object.
+Likewise, we'd like to improve the process for updating a CA certificate when expired,
+compromised or for any other reason in each route object. Instead, this document proposes a
+simpler alternative: a feature in which an administrator could define a CA bundle for verifying
+certificates from a ConfigMap consisting of the list of CA certificates, associated OpenShift
+Ingress Controller, and trusted OpenShift CA.
+
+### User Stories
+- As an OpenShift cluster administrator, I want to configure admin CA certificate bundle which
+  should be used for verifying service's certificate by the OpenShift router, instead of
+  having to configure the same as destinationCACertificate in each of the created route object.
+- As an OpenShift developer, I want to create a CA trust bundle for Ingress operator usage,
+  which includes the OpenShift service CA and the admin provided CA bundle as the default CA
+  for verifying the service's certificate when a destination CA certificate is not configured
+  for a route.
+- As an OpenShift developer, I want to update the OpenShift router certificate cache whenever
+  the OpenShift service CA certificate or the admin configured CA certificate bundle is updated
+  or deleted.
+
+### Goals
+
+- An OpenShift administrator can configure a CA certificate bundle and expect that to be used
+  for verifying the service's certificate by the OpenShift router without defining the
+  destinationCACertificate in the route object created for the service.
+- Openshift Router should be updated seamlessly when the OpenShift service CA certificate or
+  the administrator configured CA certificate bundles are modified.
+
+### Non-Goals
+
+- OpenShift administrator defined CA certificate bundle will be used only for the traffic
+  through OpenShift router and for the routes created with re-encrypt termination type, any
+  other scenario is outside the scope of this proposal.
+
```

Comment:
This proposal covers the default Ingress Controller only, and does not apply to sharded Ingress Controllers.

Reply:
Yes, the same CA bundle is made available for all IngressControllers, but the implementation is dependent on the default IngressController for reconciling and cleanup of `ingress-ca-bundle` ConfigMap.

Reply:
Why does the implementation need to watch any IngressController?  It seems to me that you need a new controller that only watches the service CA ConfigMap, custom CA ConfigMap, and `ingress-ca-bundle` ConfigMap.

Reply:
I had this misunderstanding other ingresscontroller existence won't matter when default ingresscontroller doesn't exist. 
Correction, implementation does not depend on a IngressController.